File: telegram-bot-spring-boot-starter/src/main/java/ru/tggc/telegrambotspringbootstarter/TelegramProperties.java
package ru.tggc.telegrambotspringbootstarter;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "telegram")
@Data
public class TelegramProperties {
    /**
     * Message capture mode
     */
    private Mode mode = Mode.LONG_POLLING;
    /**
     * Webhook properties
     */
    private Webhook webhook = new Webhook();

    /**
     * Telegram bot token
     */
    private String token;
    /**
     * Admin's telegram id
     */
    private String adminId;
    /**
     * Bot's Telegram ID
     */
    private String botId;

    public enum Mode {
        WEBHOOK,
        LONG_POLLING
    }

    @Data
    public static class Webhook {
        /**
         * Webhook URL
         */
        private String url;
        /**
         * Webhook path
         */
        public String path;
    }
}

File: telegram-bot-spring-boot-starter/src/main/kotlin/ru/tggc/telegrambotspringbootstarter/TelegramProperties.kt
package ru.tggc.telegrambotspringbootstarter

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "telegram")
data class TelegramProperties(
    /**
     * Message capture mode
     */
    var mode: Mode = Mode.LONG_POLLING,

    /**
     * Webhook properties
     */
    var webhook: Webhook = Webhook(),

    /**
     * Telegram bot token
     */
    var token: String? = null,

    /**
     * Admin's telegram id
     */
    var adminId: String? = null,

    /**
     * Bot's Telegram ID
     */
    var botId: String? = null,

    /**
     * List of messages base names
     */
    var baseNames: List<String> = listOf(
        "telegram/messages/messages"
    )
) {

    enum class Mode {
        WEBHOOK,
        LONG_POLLING
    }

    data class Webhook(
        /**
         * Webhook URL
         */
        var url: String? = null,

        /**
         * Webhook path
         */
        var path: String? = null
    )
}
